bottlecap/src/traces/http_utils.rs

```diff
@@ -0,0 +1,162 @@
+// Copyright 2023-Present Datadog, Inc. https://www.datadoghq.com/
+// SPDX-License-Identifier: Apache-2.0
+
+use hyper::{
+    header,
+    http::{self, HeaderMap},
+    Body, Response, StatusCode,
+};
+use log::{error, info};
+use serde_json::json;
+
+/// Does two things:
+/// 1. Logs the given message. A success status code (within 200-299) will cause an info log to be
+///    written,
+/// otherwise error will be written.
+/// 2. Returns the given message in the body of JSON response with the given status code.
+///
+/// Response body format:
+/// {
+///     "message": message
+/// }
+pub fn log_and_create_http_response(
+    message: &str,
+    status: StatusCode,
+) -> http::Result<Response<Body>> {
+    if status.is_success() {
+        info!("{message}");
+    } else {
+        error!("{message}");
+    }
+    let body = json!({ "message": message }).to_string();
+    Response::builder().status(status).body(Body::from(body))
+}
+
+/// Takes a request's header map, and verifies that the "content-length" header is present, valid,
+/// and less than the given `max_content_length`.
+///
+/// Will return None if no issues are found. Otherwise logs an error (with the given prefix) and
+/// returns and HTTP Response with the appropriate error status code.
+pub fn verify_request_content_length(
+    header_map: &HeaderMap,
+    max_content_length: usize,
+    error_message_prefix: &str,
+) -> Option<http::Result<Response<Body>>> {
```

Comment:
Should this be a Result instead of an Option?

Reply:
It can be, this is from the [mini agent](https://github.com/DataDog/libdatadog/blob/main/trace-mini-agent/src/http_utils.rs). We can tweak it though, just might make it more challenging to re-merge later if we go that route.